Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -201,130 +207,130 @@ struct SymbolGraphLoader {
+                                    return
+                                }
+                                symbolAvailability.availability[availabilityForFallbackPlatformIdx].introducedVersion = inheritedAvailability.introducedVersion
+                                return
+                            }
+                            // The symbols does not contains any information for the fallback platform
                             inheritedAvailability.domain = SymbolGraph.Symbol.Availability.Domain(rawValue: fallbackPlatform.rawValue)
-                            symbolAvailability.availability.removeAll(where: {
-                                $0.matches(fallbackPlatform)
-                            })
+                            inheritedAvailability.deprecatedVersion = inheritedAvailability.deprecatedVersion
                             symbolAvailability.availability.append(inheritedAvailability)
+                            if inheritedAvailability.introducedVersion != nil {
+                                platformsAvailableAtThisSymbol.append(fallbackPlatform.rawValue)
+                            }
                         }
                     }
-                    // Add fallback availability.
-                    for (fallbackPlatform, inheritedPlatform) in missingFallbackPlatforms {
-                        if !symbolAvailability.contains(fallbackPlatform) {
-                            for var fallbackAvailability in symbolAvailability.availability {
-                                // Add the platform fallback to the availability mixin the platform is inheriting from.
-                                // The added availability copies the entire availability information,
-                                // including deprecated and obsolete versions.
-                                if fallbackAvailability.matches(inheritedPlatform) {
-                                    fallbackAvailability.domain = SymbolGraph.Symbol.Availability.Domain(rawValue: fallbackPlatform.rawValue)
-                                    symbolAvailability.availability.append(fallbackAvailability)
-                                }
+                }
+                
+                // Add the module default availability information.
+                defaultAvailabilities.forEach { defaultAvailability in
+                    // Check that the symbol not has explicit availability for this platform already,
+                    // Check that if there was a symbolgraph for this platform, the symbol was present on it,
+                    // if not it means that the symbol is not available for the default platform.
+                    if (
+                        !platformsAvailableAtThisSymbol.contains(defaultAvailability.platformName.rawValue) &&
+                        (platformsRegistered[defaultAvailability.platformName.rawValue]?.contains(where: { symbolID == $0.precise }) ?? true)
```

Comment:
Question: would the second condition ever be false when the first one is true or is the first condition only for performance? The way I understand this if-statement is that the first condition checks if the default platform exists for the current selector's language and the second condition checks if the default platform exists for any language. 

If that's correct, we may be able to express both in a single condition without sacrificing performance. For example, we could gather the combined values from `platformsAvailableByLanguage` (from [this comment](https://github.com/swiftlang/swift-docc/pull/1065/files#r1819423447) above) into a set and check it here. 

If that works we can probably stop computing `platformsRegistered` since AFAICT this is the only place where it's used.

Reply:
> Question: would the second condition ever be false when the first one is true or is the first condition only for performance?

Yes, this can happen if the symbol is not available for a specific platform. This is not because of performance necessarily.

The first check makes sure that the symbol does not have availability annotation for that platform in the symbol graph, while the second statement checks if the symbol is part of the symbol graph of that platform.